### ISSUE TITLE
PR #12768 : le mini-GPT part a une perte de 107 la ou le notebook annonce ln(85)=4,44 — embedding lie non re-initialise

### ISSUE THREAD

**Issue by myia-ai-01** [jsboige/CoursIA#13279]
Sur la PR #12768 (`3.4-Attention-Transformer-From-Scratch.ipynb`, +2537/-5). Mécaniquement la PR est saine : 17/17 cellules code exécutées, 0 erreur, kernel `python3-coursia2`, `check_unaddressed_nits` OK. Le défaut est dans le **contenu**, et le notebook se contredit dans ses propres sorties committées.

## Le notebook prédit 4,44 et mesure 107

La cellule d'entraînement imprime son attente, puis la mesure, à trois lignes d'écart :

```
perte initiale attendue ~ ln(85) = 4.44 (prediction uniforme)
etape    1 | train 106.971 | val 107.465 |     1 s
```

**24× au-dessus de la valeur annoncée.** Un étudiant lit les deux lignes dans la même sortie.

Toute la lecture pédagogique est construite sur ce plancher, et devient fausse :

- prose (cellule markdown 28) : « La perte démarre à $\ln(|\mathcal{V}|) \approx 4{,}6$ » — elle démarre à 107. (Au passage, $\ln(85) = 4{,}44$, pas 4,6.)
- figure : `axhline(ln(V) = 4.44)` étiquetée « uniforme » ;
- prose (cellule 30) : « La ligne pointillée rappelle **le plancher de départ** : la distance parcourue depuis $\ln(V)$ est tout ce que le modèle sait. » Or le modèle **part 24× au-dessus** de cette ligne et **finit en dessous** (2,44). Ce n'est pas un plancher de départ, c'est une ligne que la courbe traverse vers le bas.

## Cause probable — liaison de poids sur un `nn.Embedding` non ré-initialisé

```python
self.emb  = nn.Embedding(taille_vocab, d)
self.tete = nn.Linear(d, taille_vocab, bias=False)
self.tete.weight = self.emb.weight          # liaison (weight tying)
```

`nn.Embedding` s'initialise par défaut en $\mathcal{N}(0,1)$, là où `nn.Linear` tire dans $\mathcal{U}(-1/\sqrt{d}, 1/\sqrt{d}) \approx \pm 0{,}079$. En liant la tête à l'embedding, la couche de sortie **hérite d'un écart-type de 1** : les logits sortent avec une amplitude sans rapport avec l'échelle attendue, d'où une entropie croisée initiale de ~107.

Le correctif est la convention GPT-2, une ligne dans `MiniGPT.__init__` :

```python
nn.init.normal_(self.emb.weight, std=0.02)
```

Après quoi la perte initiale vaut ≈ $\ln(85) = 4{,}44$ — c'est-à-dire **exactement ce que le notebook imprime déjà comme attendu**, et le récit du plancher redevient vrai. Verdict SOTA : **RECOVERABLE-LOCAL** (une ligne + ré-exécution CPU d'environ 3 min).

Ce n'est pas un détail cosmétique : le notebook enseigne le transformeur *from scratch*, et l'initialisation de l'embedding lié est précisément un des points où une implémentation maison diverge d'une implémentation correcte. Le laisser en l'état enseigne l'inverse de la leçon.

## En second lieu — le body de la PR cite des chiffres d'un run antérieur

Le body affirme « les chiffres cités en prose sont relus contre la sortie committée après exécution ». Recoupés un à un, quatre ne le sont pas :

| Body | Sortie committée |
|---|---|
| entraînement **117 s** ; notebook entier **~125 s** | `entrainement termine : 158 s sur CPU (800 etapes)` |
| chute à **train 2,46 / val 2,45** | `etape 800 \| train 2.440 \| val 2.433` (2,46/2,45 = l'étape 600) |
| perplexité finale **train 11,7 / val 11,5** | `perplexite finale : train 11.5 \| val 11.4` |
| perte initiale **4,44 = ln(85)** | `etape 1 \| train 106.971` |

Les critères d'acceptation de #12410 tiennent quand même (CPU seul, < 10 min : 158 s reste sous le seuil) — mais le body a été rédigé contre un run antérieur, et c'est le même angle mort que #13275 et #13277 ouverts cette semaine : **une ré-exécution authentique change les sorties, et la prose qui en dérive reste sur les anciennes valeurs.** Aucun garde ne l'attrape : checks verts, nits verts.

Ce qui est **juste** dans le body, vérifié : écart maison vs `torch.nn.MultiheadAttention` `2.384e-07`, 1 251 040 paramètres, vocabulaire de 85 caractères, poids de la requête 0 `[... 0.967]`.

## Attendu

1. `nn.init.normal_(self.emb.weight, std=0.02)` (ou équivalent argumenté), ré-exécution complète, sorties committées.
2. Ré-alignement de la prose des cellules 28 et 30 sur les **nouvelles** sorties — et correction du `\approx 4{,}6` en $\ln(85) = 4{,}44$.
3. Body ré-écrit contre la sortie finale, pas contre un run intermédiaire.

Si l'initialisation est **délibérée** (montrer qu'un mauvais init coûte 24× la perte attendue serait un excellent moment pédagogique), alors elle doit être **dite** : la prose actuelle affirme le contraire de ce que la sortie montre.
